.github/workflows/base-ci-goreleaser.yaml

```diff
@@ -64,9 +68,73 @@ jobs:
           go-version: '1.23'
           check-latest: true
 
-      - name: Generate the sources
+      - name: Get latest finished run ID from contrib repo build-and-test
+        id: get-run-id
+        if: inputs.latest == true
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          run_id=$(gh run list \
+            --branch main \
+            --workflow build-and-test \
+            --repo open-telemetry/opentelemetry-collector-contrib \
+            --limit 1 \
+            --status success \
+            --json databaseId \
+            --jq '.[0].databaseId' \
+          )
+          echo "Found run ID: $run_id"
```

Comment:
What happens if a run id is not found?

Reply:
There will always be a `build-and-test` run that succeeded. I double checked things like pagination and limits set up in the GH CLI and also in the GH API behind and it should always be fine.